### PR TITLE
set log format using a flag

### LINKED ISSUE #1029: better logging format
#327 forces logrus to output colored logs, which are very hard to parse using the built-in tools of log collectors like fluentd.

In case you want to keep the existing behaviour, will you be open to accept a PR to allow setting the log formatter using a flag?

### PR DESCRIPTION
**Pull request by osela**
Fixes #1029

**Description**

This PR adds a `log-format` flag to make users' life easier when using log collectors like fluentd or in other scenarios where logs are parsed.
The flag supports 3 options:
- "text" - standard text format
- "color" - colored text format (default - to keep existing behavior)
- "json"

As part of the change I introduced a new `logging` package to hold all logic that deals with logging.

As the logic is mostly inside `main` I didn't find much value for unit tests, but I can add them if you insist :)

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [x] The code flow looks good. 
- [x] Unit tests and or integration tests added.

**Release Notes**

Describe any changes here so maintainer can include it in the release notes, or delete this block.

```
- Add log-format flag
```

### CODE REVIEW

**Comment by osela** [comment]
@googlebot I signed it!

**Comment by tejal29** [comment]
@osela Please add bioler plate license code to newly added go file.

Thanks!

File: pkg/logging/logging.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			DisableColors: true,
+		}
+	case FormatColor:
+		formatter = &logrus.TextFormatter{
+			ForceColors: true,
+		}
+	case FormatJSON:
+		formatter = &logrus.JSONFormatter{}
+	default:
+		return fmt.Errorf("not a valid log format: %q", logFormat)
```
Comment (tejal29) [inline pkg/logging/logging.go]:
```suggestion
		return fmt.Errorf("not a valid log format: %q. Please specify one of (text, color, json)", logFormat)
```

Comment (osela) [reply pkg/logging/logging.go]: done, thanks

File: pkg/logging/logging.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2018 Google LLC
```
Comment (tejal29) [inline pkg/logging/logging.go]:
```suggestion
Copyright 2020 Google LLC
```

**Comment by tejal29** [comment]
restarted the Travis build. Failed due to random docker install error.

File: pkg/logging/logging.go (hunk 2 above)
Comment (osela) [reply pkg/logging/logging.go]: oops...
